src/opendap/build_dmrpp/BuildDmrppDispatchHandler.java

```diff
@@ -0,0 +1,238 @@
+    private Logger log;
+    private boolean _initialized;
+    private String _prefix = "build_dmrpp";
+    private BuildDmrppBesApi _besApi;
+
+    private static final String d_landingPage="/docs/ngap/ngap.html";
+
+    public BuildDmrppDispatchHandler() {
+        super();
+        log = org.slf4j.LoggerFactory.getLogger(getClass());
+        _initialized = false;
+        _besApi = null;
+    }
+
+    @Override
+    public void init(HttpServlet servlet, Element config) throws Exception {
+        init(servlet,config,null);
+    }
+
+
+    @Override
+    public void init(HttpServlet servlet, Element config, BesApi ignored) throws Exception {
+
+        if(_initialized)
+            return;
+
+        ingestPrefix(config);
+
+        _besApi = new BuildDmrppBesApi(_prefix);
+        super.init(servlet, config, _besApi);
+        _initialized=true;
+    }
+
+    @Override
+    public boolean requestDispatch(HttpServletRequest request,
+                                   HttpServletResponse response,
+                                   boolean sendResponse)
+            throws Exception {
+
+        User user = new User(request);
+        QueryParameters qp = new QueryParameters(request);
+
+        String invocation = request.getRequestURL().toString();
+        String qs = request.getQueryString();
+        if(qs!=null)
+            invocation += "?" + qs;
+        log.debug("invocation:    "+invocation);
+
+        String relativeURL = ReqInfo.getLocalUrl(request);
+        log.debug("relativeURL:    "+relativeURL);
+
+        while(relativeURL.startsWith("/") && relativeURL.length()>1)
+            relativeURL = relativeURL.substring(1,relativeURL.length());
+        boolean itsJustThePrefixWithoutTheSlash = _prefix.substring(0,_prefix.lastIndexOf("/")).equals(relativeURL);
+        boolean itsJustThePrefix = _prefix.equals(relativeURL);
+        boolean isMyRequest = itsJustThePrefixWithoutTheSlash || relativeURL.startsWith(_prefix);
+
+        if(isMyRequest) {
+            if (sendResponse) {
+
+                if(itsJustThePrefixWithoutTheSlash ){
+                    response.sendRedirect(_prefix);
+                    log.debug("Sent redirect to service prefix: "+_prefix);
+                    return true;
+                }
+
+                reqCounter.incrementAndGet();
+                if(itsJustThePrefix){
+                    sendSimpleNgapLandingPage(response);
+                }
+                else {
+                    log.info("Sending build_dmrpp Response");
+
+                    if(relativeURL.startsWith(_prefix)){
+                        relativeURL = relativeURL.substring(_prefix.length());
+                    }
+
+                    Document build_dmrpp_cmd;
+                    build_dmrpp_cmd = _besApi.getBuildDmrppDocument(user, relativeURL, qp, invocation);
+
+                    XMLOutputter xmlo = new XMLOutputter(Format.getPrettyFormat());
+                    log.error(xmlo.outputString(build_dmrpp_cmd));
```

Comment:
This:
` log.error(xmlo.outputString(build_dmrpp_cmd));`
Means that it will be written to the log every time. (Because our configuration always writes errors to the OLFS logs) 
I think it was probably a convenience so that the result would show up always. Probably it should be:
`log.info(xmlo.outputString(build_dmrpp_cmd));`
And that's redundant as well since it's going in the BesCommand log as well.

Reply:
Which of the two options is preferable?  Changing to log.info() or removing completely?

Reply:
remove, please.